app/Config.hs

```diff
@@ -16,10 +17,8 @@ import System.Hapistrano.Commands
 data Config = Config
   { configDeployPath :: !(Path Abs Dir)
     -- ^ Top-level deploy directory on target machine
-  , configHost :: !(Maybe String)
-    -- ^ Host to deploy to. If missing, localhost will be assumed.
-  , configPort :: !Word
-    -- ^ SSH port number to use, may be omitted
+  , configHosts :: ![(String, Word)]
```

Comment:
@mrkkrp what about the scenario when a user add the configuration to the same server twice? Then hapistrano is going to trigger two deployments to the same instance concurrently which make cause some race condition issues. I wonder if we could use a `Map String Word` to prevent this to happen, or maybe validate the hosts before starting the deployment process?